src/mpt.rs

```diff
@@ -132,6 +154,7 @@ impl MptCircuitConfig {
                 for offset in 1 + n_assigned_rows..n_rows {
```

Comment:
In order to check that a padding row is assigned, the assert above is off by 2 (1 first row + 1 final row). It should be:

                assert!(
                    1 + n_assigned_rows < n_rows,
                    "mpt circuit requires {n_assigned_rows}+2 rows > limit of {n_rows} rows"
                );